### ISSUE TITLE
Windows 7 pip install error

### ISSUE THREAD

**Issue by MikeyR16** [sopel-irc/sopel#811]
I have an error when I try to install it through pip, not sure hwy some of the lines are missing. 
Using cmd prompt, win 7 64 bit, Python 2.7.8 anaconda 2.1 (64 bit)
```python
lace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\mikeyr\appda
\local\temp\pip-8kmjya-record\install-record.txt --single-version-externally-m
aged --compile:
    running install
    running build
    running build_py
    running build_scripts
    running install_lib
    copying build\lib\willie\bot.py -> C:\Anaconda\Lib\site-packages\willie
    copying build\lib\willie\config\__init__.py -> C:\Anaconda\Lib\site-packag
\willie\config
    copying build\lib\willie\coretasks.py -> C:\Anaconda\Lib\site-packages\wil
e
    copying build\lib\willie\db.py -> C:\Anaconda\Lib\site-packages\willie
    copying build\lib\willie\formatting.py -> C:\Anaconda\Lib\site-packages\wi
ie
    copying build\lib\willie\irc.py -> C:\Anaconda\Lib\site-packages\willie
    copying build\lib\willie\logger.py -> C:\Anaconda\Lib\site-packages\willie
    copying build\lib\willie\module.py -> C:\Anaconda\Lib\site-packages\willie
    copying build\lib\willie\modules\admin.py -> C:\Anaconda\Lib\site-packages
illie\modules
    copying build\lib\willie\modules\adminchannel.py -> C:\Anaconda\Lib\site-p
kages\willie\modules
    copying build\lib\willie\modules\announce.py -> C:\Anaconda\Lib\site-packa
s\willie\modules
    copying build\lib\willie\modules\bugzilla.py -> C:\Anaconda\Lib\site-packa
s\willie\modules
    copying build\lib\willie\modules\calc.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\chanlogs.py -> C:\Anaconda\Lib\site-packa
s\willie\modules
    copying build\lib\willie\modules\clock.py -> C:\Anaconda\Lib\site-packages
illie\modules
    copying build\lib\willie\modules\countdown.py -> C:\Anaconda\Lib\site-pack
es\willie\modules
    copying build\lib\willie\modules\currency.py -> C:\Anaconda\Lib\site-packa
s\willie\modules
    copying build\lib\willie\modules\dice.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\etymology.py -> C:\Anaconda\Lib\site-pack
es\willie\modules
    copying build\lib\willie\modules\find.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\find_updates.py -> C:\Anaconda\Lib\site-p
kages\willie\modules
    copying build\lib\willie\modules\github.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\help.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\ip.py -> C:\Anaconda\Lib\site-packages\wi
ie\modules
    copying build\lib\willie\modules\ipython.py -> C:\Anaconda\Lib\site-packag
\willie\modules
    copying build\lib\willie\modules\isup.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\lmgtfy.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\meetbot.py -> C:\Anaconda\Lib\site-packag
\willie\modules
    copying build\lib\willie\modules\movie.py -> C:\Anaconda\Lib\site-packages
illie\modules
    copying build\lib\willie\modules\ping.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\radio.py -> C:\Anaconda\Lib\site-packages
illie\modules
    copying build\lib\willie\modules\rand.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\reddit.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\reload.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\remind.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\rss.py -> C:\Anaconda\Lib\site-packages\w
lie\modules
    copying build\lib\willie\modules\safety.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\search.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\seen.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\spellcheck.py -> C:\Anaconda\Lib\site-pac
ges\willie\modules
    copying build\lib\willie\modules\tell.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\tld.py -> C:\Anaconda\Lib\site-packages\w
lie\modules
    copying build\lib\willie\modules\translate.py -> C:\Anaconda\Lib\site-pack
es\willie\modules
    copying build\lib\willie\modules\unicode_info.py -> C:\Anaconda\Lib\site-p
kages\willie\modules
    copying build\lib\willie\modules\units.py -> C:\Anaconda\Lib\site-packages
illie\modules
    copying build\lib\willie\modules\uptime.py -> C:\Anaconda\Lib\site-package
willie\modules
    copying build\lib\willie\modules\url.py -> C:\Anaconda\Lib\site-packages\w
lie\modules
    copying build\lib\willie\modules\version.py -> C:\Anaconda\Lib\site-packag
\willie\modules
    copying build\lib\willie\modules\weather.py -> C:\Anaconda\Lib\site-packag
\willie\modules
    copying build\lib\willie\modules\wikipedia.py -> C:\Anaconda\Lib\site-pack
es\willie\modules
    copying build\lib\willie\modules\wiktionary.py -> C:\Anaconda\Lib\site-pac
ges\willie\modules
    copying build\lib\willie\modules\xkcd.py -> C:\Anaconda\Lib\site-packages\
llie\modules
    copying build\lib\willie\modules\__init__.py -> C:\Anaconda\Lib\site-packa
s\willie\modules
    copying build\lib\willie\test_tools.py -> C:\Anaconda\Lib\site-packages\wi
ie
    copying build\lib\willie\tools\calculation.py -> C:\Anaconda\Lib\site-pack
es\willie\tools
    copying build\lib\willie\tools\jobs.py -> C:\Anaconda\Lib\site-packages\wi
ie\tools
    copying build\lib\willie\tools\time.py -> C:\Anaconda\Lib\site-packages\wi
ie\tools
    copying build\lib\willie\tools\__init__.py -> C:\Anaconda\Lib\site-package
willie\tools
    copying build\lib\willie\trigger.py -> C:\Anaconda\Lib\site-packages\willi
    copying build\lib\willie\web.py -> C:\Anaconda\Lib\site-packages\willie
    copying build\lib\willie\__init__.py -> C:\Anaconda\Lib\site-packages\will

    byte-compiling C:\Anaconda\Lib\site-packages\willie\bot.py to bot.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\config\__init__.py to
init__.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\coretasks.py to coreta
s.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\db.py to db.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\formatting.py to forma
ing.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\irc.py to irc.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\logger.py to logger.py
    byte-compiling C:\Anaconda\Lib\site-packages\willie\module.py to module.py
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\admin.py to ad
n.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\adminchannel.p
to adminchannel.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\announce.py to
nnounce.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\bugzilla.py to
ugzilla.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\calc.py to cal
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\chanlogs.py to
hanlogs.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\clock.py to cl
k.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\countdown.py t
countdown.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\currency.py to
urrency.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\dice.py to dic
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\etymology.py t
etymology.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\find.py to fin
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\find_updates.p
to find_updates.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\github.py to g
hub.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\help.py to hel
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\ip.py to ip.py
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\ipython.py to
ython.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\isup.py to isu
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\lmgtfy.py to l
tfy.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\meetbot.py to
etbot.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\movie.py to mo
e.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\ping.py to pin
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\radio.py to ra
o.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\rand.py to ran
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\reddit.py to r
dit.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\reload.py to r
oad.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\remind.py to r
ind.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\rss.py to rss.
c
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\safety.py to s
ety.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\search.py to s
rch.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\seen.py to see
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\spellcheck.py
 spellcheck.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\tell.py to tel
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\tld.py to tld.
c
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\translate.py t
translate.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\unicode_info.p
to unicode_info.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\units.py to un
s.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\uptime.py to u
ime.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\url.py to url.
c
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\version.py to
rsion.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\weather.py to
ather.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\wikipedia.py t
wikipedia.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\wiktionary.py
 wiktionary.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\xkcd.py to xkc
pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\modules\__init__.py to
_init__.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\test_tools.py to test_
ols.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\tools\calculation.py t
calculation.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\tools\jobs.py to jobs.
c
    byte-compiling C:\Anaconda\Lib\site-packages\willie\tools\time.py to time.
c
    byte-compiling C:\Anaconda\Lib\site-packages\willie\tools\__init__.py to _
nit__.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\trigger.py to trigger.
c
    byte-compiling C:\Anaconda\Lib\site-packages\willie\web.py to web.pyc
    byte-compiling C:\Anaconda\Lib\site-packages\willie\__init__.py to __init_
pyc
    running install_egg_info
    running egg_info
    writing requirements to willie.egg-info\requires.txt
    writing willie.egg-info\PKG-INFO
    writing top-level names to willie.egg-info\top_level.txt
    writing dependency_links to willie.egg-info\dependency_links.txt
    warning: manifest_maker: standard file '-c' not found

    reading manifest file 'willie.egg-info\SOURCES.txt'
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "c:\users\mikeyr\appdata\local\temp\pip-build-xclgam\willie\setup.p
, line 59, in <module>
        do_setup()
      File "c:\users\mikeyr\appdata\local\temp\pip-build-xclgam\willie\setup.p
, line 49, in do_setup
        setup(**setup_args)
      File "C:\Anaconda\lib\distutils\core.py", line 151, in setup
        dist.run_commands()
      File "C:\Anaconda\lib\distutils\dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "C:\Anaconda\lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\install.py", line 61, in run
      File "C:\Anaconda\lib\distutils\command\install.py", line 575, in run
        self.run_command(cmd_name)
      File "C:\Anaconda\lib\distutils\cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "C:\Anaconda\lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\install_egg_info.py", line 33, in run
      File "C:\Anaconda\lib\distutils\cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "C:\Anaconda\lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\egg_info.py", line 168, in run
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\egg_info.py", line 193, in find_sources
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\egg_info.py", line 277, in run
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\egg_info.py", line 313, in add_defaults
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\sdist.py", line 249, in read_manifest
      File "C:\Anaconda\lib\site-packages\setuptools-5.8-py2.7.egg\setuptools\
mmand\egg_info.py", line 218, in append
      File "C:\Anaconda\lib\distutils\util.py", line 124, in convert_path
        raise ValueError, "path '%s' cannot be absolute" % pathname
    ValueError: path '/tmp/tmpeAhjCF/willie' cannot be absolute

    ----------------------------------------
Command "C:\Anaconda\python.exe -c "import setuptools, tokenize;__file__='c:\\
ers\\mikeyr\\appdata\\local\\temp\\pip-build-xclgam\\willie\\setup.py';exec(co
ile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __
le__, 'exec'))" install --record c:\users\mikeyr\appdata\local\temp\pip-8kmjya
ecord\install-record.txt --single-version-externally-managed --compile" failed
ith error code 1 in c:\users\mikeyr\appdata\local\temp\pip-build-xclgam\willie
```

**Comment by embolalia** [comment]
Your error message looks like it's missing a few characters from each line. You also have it rendering as python code, but it's just plaintext, so there's all sorts of distracting formatting going on.

At a guess, from `ValueError: path '/tmp/tmpeAhjCF/willie' cannot be absolute`, the problem is [this stuff](https://github.com/embolalia/willie/blob/master/setup.py#L26-L28). I don't know why it would be generating a bad path, though. I don't have Windows to test on, though (@tyrope?).

**Comment by tyrope** [comment]
Just tested that function, returned D:\Temp\somerandomcharacters so if that is broken, its because of reasons outside our scope.

Could you run you python interpreter with the following and post the result?

import tempfile;print tempfile.mkdtemp()

**Comment by MikeyR16** [comment]
@tyrope I started python in my cmd and executed the following line 
```python
import tempfile;print tempfile.mkdtemp()
```
The results are: c:\users\mikeyr\appdata\local\temp\tmpdd1tgq

Just an empty directory: http://i.imgur.com/4lSH2pk.png

**Comment by embolalia** [comment]
I'm not familiar at all with Anaconda, and at a glance I'm not sure what the point of using it is here. Try running it without Anaconda and see if that helps. @tyrope, can you check if `pip install willie` works on a normal Windows Python installation?

**Comment by quintopia** [comment]
I can confirm this bug on a normal windows python install. the failure looks identical. the relevant lines are: 
      File "C:\Python27\lib\distutils\util.py", line 124, in convert_path
        raise ValueError, "path '%s' cannot be absolute" % pathname
    ValueError: path '/tmp/tmpeAhjCF/willie' cannot be absolute

**Comment by tyrope** [comment]
@embolalia My python install apparently didn't come with pip. Cannot reproduce.

**Comment by hotsammysliz** [comment]
I can also confirm I am getting the same identical error as @tyrope and @quintopia and I am installing using a normal windows python 2.7 pip install.

**Comment by Phillammon** [comment]
The me too's probably aren't helping, but have also reproduced here. I have the same string of random letters too, which may or may not be relevant.

**Comment by embolalia** [comment]
It's always `'/tmp/tmpeAhjCF/willie'`? Is this true for everyone? I think this is going to require someone with Windows to do some deep digging and figure out where that pathname is *actually* coming from. If it's always the same, it's probably hardcoded somehow somewhere, but I don't know why or where, and grep isn't helping me.

**Comment by Phillammon** [comment]
Spitballing ideas here, could it be related to that old "random numbers in windows aren't actually random" thing?

**Comment by joshood** [comment]
The lines referred to by @embolalia work properly, as does cloing the repo and running
 `python setup.py install`

Something's up with the way this release was packaged. I downloaded the [tarball](https://pypi.python.org/pypi/willie/5.3.0) from PyPi and found the offending string in willie-5.3.0\willie.egg-info\SOURCES.txt, line 5.

I don't know much about eggs or how to build them, but I think creating a new one can fix this.

**Comment by Kinematics** [comment]
Deleted the line joshood found, and it allowed willie to actually be installed.  However I'm running into numerous errors when trying to run it, so there may be other issues with the installer package.

**Comment by Chris-Su** [comment]
The problem lies in the "Python34\Lib\distutils\utils.py" module. 
The documentation "Return 'pathname' as a name that will work on the native filesystem,
    i.e. split it on '/' and put it back together again using the current
    directory separator.  Needed because filenames in the setup script are
    always supplied in Unix style, and have to be converted to the local
    convention before we can actually use them in the filesystem.  **Raises
    ValueError on non-Unix-ish systems if 'pathname' either starts or
    ends with a slash.**"

**Comment by embolalia** [comment]
The issue is with the hacky solution to getting the run script installed in the right place with the right name. #622 recommended switching to a setuptools entry point, which is definitely cleaner, but I turned it down because it would be painful on platforms that have to manually bootstrap setuptools. Apparently it's going to be painful anyway, though, so I might change my mind on that.

**Comment by embolalia** [comment]
This was fixed by 9375904.
